zingolib/src/wallet/tx_map_and_maybe_trees/trait_walletread.rs

```diff
@@ -0,0 +1,382 @@
+    keys::UnifiedFullViewingKey,
+};
+use zcash_primitives::consensus::BlockHeight;
+use zip32::AccountId;
+
+use crate::{error::ZingoLibError, wallet::notes::query::OutputQuery};
+
+use super::TxMapAndMaybeTrees;
+
+/// This is a facade for using LRZ traits. In actuality, Zingo does not use multiple accounts in one wallet.
+pub struct ZingoAccount(AccountId, UnifiedFullViewingKey);
+
+impl Account<AccountId> for ZingoAccount {
+    fn id(&self) -> AccountId {
+        self.0
+    }
+
+    fn source(&self) -> zcash_client_backend::data_api::AccountSource {
+        unimplemented!()
+    }
+
+    fn ufvk(&self) -> Option<&UnifiedFullViewingKey> {
+        Some(&self.1)
+    }
+
+    fn uivk(&self) -> zcash_keys::keys::UnifiedIncomingViewingKey {
+        unimplemented!()
+    }
+}
+
+/// some of these functions, initially those required for calculate_transaction, will be implemented
+impl WalletRead for TxMapAndMaybeTrees {
+    type Error = ZingoLibError;
+    type AccountId = AccountId;
+    type Account = ZingoAccount;
+
+    /// Returns the account corresponding to a given [`UnifiedFullViewingKey`], if any.
+    /// IMPL: partially implemented. zingo uses the default account. when we expand account functionality, this will be updated
+    fn get_account_for_ufvk(
+        &self,
+        ufvk: &UnifiedFullViewingKey,
+    ) -> Result<Option<Self::Account>, Self::Error> {
+        // todo we could assert that the ufvk matches, or return error.
+        Ok(Some(ZingoAccount(AccountId::ZERO, ufvk.clone())))
+    }
+
+    /// Returns the default target height (for the block in which a new
+    /// transaction would be mined) and anchor height (to use for a new
+    /// transaction), given the range of block heights that the backend
+    /// knows about.
+    ///
+    /// This will return `Ok(None)` if no block data is present in the database.
+    /// IMPL: fully implemented. the target height is always the next block, and the anchor is a variable depth below.
+    /// IMPL: tested
+    fn get_target_and_anchor_heights(
+        &self,
+        min_confirmations: std::num::NonZeroU32,
+    ) -> Result<
+        Option<(
+            zcash_primitives::consensus::BlockHeight,
+            zcash_primitives::consensus::BlockHeight,
+        )>,
+        Self::Error,
+    > {
+        match self.witness_trees.as_ref() {
+            Some(trees) => {
+                let highest_block_height =
+                    match trees.witness_tree_orchard.store().max_checkpoint_id() {
+                        Ok(height) => height,
+                        // Infallible
+                        Err(e) => match e {},
+                    };
+
+                Ok(highest_block_height.map(|height| {
+                    (
+                        height + 1,
+                        BlockHeight::from_u32(std::cmp::max(
+                            1,
+                            u32::from(height).saturating_sub(u32::from(min_confirmations)),
```

Comment:
This is off-by-one of the desired result, I believe. Since min_confirmations is a NonZeroU32, in order to have the anchor height be the latest block, min_confirmations of 1 should result in an anchor height that's one below the target height.